src/Scriban/Parsing/Parser.Statements.cs

```diff
@@ -52,6 +52,18 @@ private ScriptBlockStatement ParseBlockStatement(ScriptNode parentStatement, boo
 
                     if (hasEnd)
                     {
+                        if (_blockLevel == 1)
+                        {
+                            if (_isLiquid)
+                            {
+                                var syntax = ScriptSyntaxAttribute.Get(parentStatement);
+                                LogError(parentStatement, parentStatement?.Span ?? CurrentSpan, $"Found <end> statement `end{syntax.TypeName}` without a corresponding beginning of a block");
```

Comment:
The error location seems to be off above. The log should probably use the `statement` instead of `parentStatement` (for both Scriban and Liquid) to get the correct Span location. In the case of an `end`, the statement should be the end statement.

Reply:
Fixed by 7804a97